torch/onnx/_internal/exporter.py

```diff
@@ -410,7 +439,137 @@ def save(
             serializer.serialize(self, destination)
 
 
-class Exporter(abc.ABC):
+class FXGraphExtractor(abc.ABC):
+    """Abstract interface for FX graph extractor engines.
+    This class isolates FX extraction logic from the rest of the export logic.
+    That allows a single ONNX exporter that can leverage different FX graphs."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.input_adapter: InputAdapter = InputAdapter()
+        self.output_adapter: OutputAdapter = OutputAdapter()
+
+    @_beartype.beartype
+    def _export_fx_to_onnx(
```

Comment:
This feels weird to be here. This class is `FXGraphExtractor`, but this method finishes everything off with a onnx ModelProto. Feels this method belongs to `Exporter` (or `DynamoExporter(Exporter)` for the sake of import dep problems).

Reply:
To be honest, there is no right place on the repo for `_export_fx_to_onnx`. Next round of design can split it in several methods in their correct places.

Reply:
it is the "fx passes" + "conversion to onnxscript" part. Again as discussed my hunch is they should be the general part of exporter. I'm just surprised that it is moved from `DynamoExporter` to `Extractor`. So now all the work e2e is completed in `Extractor`, that feels like just a rename of the previous `FXExporter` though. Anyway won't block on this too, as you mentioned there will be follow up designs.

Reply:
`class Exporter` as proposed by Aaron is just a thread-safety container to keep user model+data+export options together so that we could run exporter in parallel (in theory). It was never meant to have hold code. The fact we did specialize and put a lot of helpers on it is the actual infraction we need to address.

`_export_fx_to_onnx` does too much (passes, graph building and protobuf serialization) and not even `Exporter` is the right place to it. In next steps, we will be able to break all three tasks in 3 different blocks, and neither `Exporter` nor `FXGraphExtractor` will need to hold any of it.

Reply:
I agree these are 3 separate tasks. In my understanding the `Exporter`, instead of `FXGraphExtractor`, should drive these components.

Reply:
Yeah, agreed. `Exporter` must drive the 3 classes that are to be born.
`_export_fx_to_onnx` will be gone and not inside fx_tracer very soon